backend/pkg/service/service.go

```diff
@@ -269,3 +268,128 @@ func (b *backendService) ListType(ctx context.Context, in *pb.ListTypeRequest, o
 	log.Infof("Types:%+v\n", out.Types)
 	return nil
 }
+
+func (b *backendService) CreateTier(ctx context.Context, in *pb.CreateTierRequest, out *pb.CreateTierResponse) error {
+	log.Info("Received CreateTier request.")
+	tier := &model.Tier{
+		Name:     in.Tier.Name,
+		TenantId: in.Tier.TenantId,
+		Backends: in.Tier.Backends,
+	}
+	res, err := db.Repo.CreateTier(ctx, tier)
+	if err != nil {
+		log.Errorf("Failed to create tier: %v", err)
+		return err
+	}
+	out.Tier = &pb.Tier{
+		Id:       res.Id.Hex(),
+		Name:     res.Name,
+		TenantId: res.TenantId,
+		Backends: res.Backends,
+	}
+	log.Info("Create tier successfully.")
+	return nil
+}
+
+func (b *backendService) UpdateTier(ctx context.Context, in *pb.UpdateTierRequest, out *pb.UpdateTierResponse) error {
+	log.Info("Received UpdateTier request.")
+//
+	res,err:= db.Repo.GetTier(ctx,in.Id)
+
+	//to delete backends
+	var delBackends []string
+	for _,backendId:= range res.Backends{
+		found:=false
+		for _,bcknd:= range in.DeleteBackends{
+			if(backendId==bcknd){
+				found=true
+			}
+		}
+		if found==false{
+			delBackends= append(delBackends,backendId)
+		}
+	}
+
+	//add backends to be added
+	res.Backends= delBackends
```

Comment:
Can you check if addition list and deletion list are updated properly, here?

Reply:
resolved